lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3810,72 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		self.channel_transaction_parameters.channel_type_features = self.channel_type.clone();
 		Ok(())
 	}
+
+	// /// Splice process starting; update state; update capacity, state, reset funding tx
+	// #[cfg(splicing)]
+	// pub(crate) fn splice_start<L: Deref>(&mut self, is_outgoing: bool, logger: &L) where L::Target: Logger {
+	// 	// Set state, by this point handshake is complete
+	// 	self.channel_state = ChannelState::NegotiatingFunding(NegotiatingFundingFlags::OUR_INIT_SENT | NegotiatingFundingFlags::THEIR_INIT_SENT);
+
+	// 	log_info!(logger,
+	// 		"Splicing process started, new channel value {}, outgoing {}, channel_id {}",
+	// 		self.channel_value_satoshis, is_outgoing, self.channel_id,
+	// 	);
+	// }
+
+	/// Get the splice message that can be sent during splice initiation.
+	#[cfg(splicing)]
+	pub fn get_splice_init(&self, our_funding_contribution_satoshis: i64, signer_provider: &SP,
+		funding_feerate_perkw: u32, locktime: u32,
+	) -> msgs::SpliceInit {
+		if !self.is_outbound() {
+			panic!("Tried to initiate a splice on an inbound channel!");
+		}
+
+		// At this point we are not committed to the new channel value yet, but the funding pubkey
+		// depends on the channel value, so we create here a new funding pubkey with the new
+		// channel value.
+		// Note that channel_keys_id is supposed NOT to change
+		let funding_pubkey = {
+			// TODO: Funding pubkey generation requires the post channel value, but that is not known yet,
+			// the acceptor contribution is missing. There is a need for a way to generate a new funding pubkey,
+			// not based on the channel value
```

Comment:
Seems the funding pubkey derivation is not dependent on the amount in `InMemorySigner`. But it is passed to its constructor and used later when signing. @TheBlueMatt Seems will need to refactor our signing code to take `channel_value_satoshis` when signing instead. Do you see any issues with that?

Reply:
I dropped creating new funding pubkey (as it was re-creating the same value anyways), therefore no need for signer generation.